### ISSUE TITLE
Custom tag self reference does not work

### ISSUE THREAD

**Issue by saraxu** [marko-js/marko#69]
I created a custom tag component, app-fieldset, which can include other custom tags, eg. app-text-field.
It works fine if the inside custom tag are other types, but I get error if it refers itself.

Template.marko in app-fieldset:

`...
<if test="fType === 'text'">
    <app-text-field viewModel="field" />
 </if>
 <else-if test="fType === 'fieldset'">
   <app-fieldset viewModel="field" /> //refers self 
 </else-if>
`

The error in console is:
Thu, 30 Apr 2015 04:14:22 GMT uncaughtException Invalid tag handler: [object Object]
Error: Invalid tag handler: [object Object]
    at Function.deferredRenderer.renderer (...../node_modules/marko/runtime/helpers.js:35:19)
    at deferredRenderer (..../node_modules/marko/runtime/helpers.js:27:26)
    at module.exports.t (..../node_modules/marko/runtime/helpers.js:181:9)
    at ..../src/components/app-fieldset/template.marko.js:59:15

I used the latest version of marko: "2.3.2" per https://github.com/raptorjs/marko/issues/66
but seems not work for this case.

Thanks,
Sara

**Comment by patrick-steele-idem** [comment]
Hi Sara,

Does your `<app-fieldset>` custom tag have a `renderer.js`? If so, which syntax of the following does your `renderer.js` use?

__Option 1:__

```javascript
exports.renderer = function(input, out) { ... }
```

__Option 2:__

```javascript
module.exports = function renderer(input, out) { ... }
```

If you have a circular dependency you need to use __Option 1__ since it allows for a partially loaded module to be returned. It is recommended that you always use the syntax shown in __Option 1__. If you do not have a `renderer.js` then I'll probably need more details to reproduce the problem.

Thanks,
Patrick

**Comment by saraxu** [comment]
Hi Patrick,

Yes, it has renderer.js and use option 1.

Thanks,
Sara

**Comment by patrick-steele-idem** [comment]
This issue was fixed as part of the fixes for #66. Closing this issue, but please let me know if you are still facing problems. Thanks.
